logdetective/logdetective.py

```diff
@@ -2,248 +2,32 @@
-        url (str): The URL of the language model to be downloaded.
-
-    Returns:
-        str: The local file path of the downloaded language model.
-    """
-    path = os.path.join(
-        os.path.expanduser(CACHE_LOC), url.split('/')[-1])
-
-    LOG.info("Downloading model from %s to %s", url, path)
-    if not os.path.exists(path):
-        if verbose:
-            path, _status = urlretrieve(url, path, MyProgressBar())
-        else:
-            path, _status = urlretrieve(url, path)
-
-    return path
-
-
-def process_log(log: str, model: Llama) -> str:
-    """
-    Processes a given log using the provided language model and returns its summary.
-
-    Args:
-        log (str): The input log to be processed.
-        model (Llama): The language model used for processing the log.
-
-    Returns:
-        str: The summary of the given log generated by the language model.
-    """
-    return model(PROMPT_TEMPLATE.format(log), max_tokens=0)["choices"][0]["text"]
-
-
-def retrieve_log_content(log_path):
-    """Get content of the file on the log_path path."""
-    parsed_url = urlparse(log_path)
-    log = ""
-
-    if not parsed_url.scheme:
-        if not os.path.exists(log_path):
-            raise ValueError(f"Local log {log_path} doesn't exist!")
-
-        with open(log_path, "rt") as f:
-            log = f.read()
-
-    else:
-        log = requests.get(log_path, timeout=60).text
-
-    return log
-
-
 def main():
     """Main execution function."""
     parser = argparse.ArgumentParser("logdetective")
-    parser.add_argument("url", type=str, default="")
-    parser.add_argument("-M", "--model", type=str, default=DEFAULT_ADVISOR)
-    parser.add_argument("-S", "--summarizer", type=str, default="drain")
-    parser.add_argument("-N", "--n_lines", type=int, default=5)
+    parser.add_argument("file", type=str,
+                        default="", help="The URL or path to the log file to be analyzed.")
+    parser.add_argument("-M", "--model", help="The path or URL of the language model for analysis.",
+                        type=str, default=DEFAULT_ADVISOR)
+    parser.add_argument("-S", "--summarizer", type=str, default="drain",
+                        help="Choose between LLM and Drain template miner as the log summarizer.\
+                                LLM must be specified as path to a model, URL or local file.")
+    parser.add_argument("-N", "--n_lines", type=int,
+                        default=8, help="The number of lines per chunk for LLM analysis.\
+                            This only makes sense when you are summarizing with LLM.")
+    parser.add_argument("-C", "--n_clusters", type=int, default=8,
+                        help="Number of clusters for Drain to organize log chunks into.\
+                            This only makes sense when you are summarizing with Drain")
     parser.add_argument("-v", "--verbose", action='count', default=0)
     parser.add_argument("-q", "--quiet", action='store_true')
```

Comment:
we need to get logging setup to the server as well

Reply:
You mean the silent option? I was thinking about that and decided against it. The major reason for quiet mode is that all the output pollutes the terminal and makes things difficult to read. But if it's going to run as a server, we don't need to worry about that.

Especially if it's going to be containerized, because in that case the only logs stored for the pod will all belong to our service, and will be easy to filter.

Reply:
sorry, I commented on the wrong line

my point is that there is no logger handler for logdetective module when running directly from CLI using fastapi so there are no logs printed in the console; I manually added one while playing with this PR so that I could see what's happening